src/auto-complete.jsx

```diff
@@ -349,8 +355,9 @@ const AutoComplete = React.createClass({
     };
 
     const requestsList = [];
+    const maxSearchResults = this.props.maxSearchResults;
 
-    this.props.dataSource.map((item) => {
+    for (let item of this.props.dataSource) {
```

Comment:
@oliviertassinari sorry I didn't use normal `for` loop, I used `every` function.
* Using `every`, we'll have very less changes in the PR
* `every`doesn't required any polyfills
* Normal `for` loop has some boilerplate code
```
for(var i = 0; i < this.props.dataSource.length; i++){
	var item = this.props.dataSource[i];
	//actual code
}
```